router.go

```diff
@@ -13,10 +13,12 @@ import (
 // directly means we'd get a collision with any other package that does the same.
 // https://play.golang.org/p/MxhRiL37R-9
 type routerContextKeyType struct{}
+type routerPathContextKeyType struct{}
 
 var (
-	routerContextKey   = routerContextKeyType{}
-	routerComponentsRe = regexp.MustCompile(`(?:^|/)(\*\w*|:\w+)`)
+	routerContextKey     = routerContextKeyType{}
+	routerPathContextKey = routerPathContextKeyType{}
```

Comment:
```suggestion
	routerRequestPatternContextKey = routerRequestPatternContextKeyType{}
```

Optional: This might be clearer to distinguish between path and pattern?